iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ParameterDeclarationImpl.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.iac.arm.tree.api.Expression;
+import org.sonar.iac.arm.tree.api.Identifier;
+import org.sonar.iac.arm.tree.api.NumericLiteral;
+import org.sonar.iac.arm.tree.api.ParameterDeclaration;
+import org.sonar.iac.arm.tree.api.ParameterType;
+import org.sonar.iac.arm.tree.api.StringLiteral;
+import org.sonar.iac.arm.tree.api.bicep.InterpolatedString;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
+import org.sonar.iac.common.api.tree.TextTree;
+import org.sonar.iac.common.api.tree.Tree;
+
+import static org.sonar.iac.arm.tree.impl.json.ArmHelper.addChildrenIfPresent;
+
+public class ParameterDeclarationImpl extends AbstractArmTreeImpl implements ParameterDeclaration {
+
+  private final SyntaxToken keyword;
+  private final Identifier name;
+  @Nullable
+  private final StringLiteral typeExpression;
+  @Nullable
+  private final SyntaxToken resource;
+  @Nullable
+  private final InterpolatedString typeInterp;
+  @Nullable
+  private final SyntaxToken equ;
+  @Nullable
+  private final Expression defaultValue;
+
+  public ParameterDeclarationImpl(SyntaxToken keyword, Identifier name, @Nullable StringLiteral typeExpression, @Nullable SyntaxToken equ, @Nullable Expression defaultValue) {
+    this.keyword = keyword;
+    this.name = name;
+    this.typeExpression = typeExpression;
+    this.equ = equ;
+    this.defaultValue = defaultValue;
+    this.resource = null;
+    this.typeInterp = null;
+  }
+
+  public ParameterDeclarationImpl(SyntaxToken keyword, Identifier name, @Nullable SyntaxToken resource, @Nullable InterpolatedString typeInterp, @Nullable SyntaxToken equ,
```

Comment:
Nitpic. Lines 55 and 65-66 are long, maybe they will look better when each argument will be in a separate line?